test.spec.js

```diff
@@ -118,6 +118,51 @@ module.exports = {
     });
   },
 
+  test_hash_negative_time_cost: function (assert) {
+    "use strict";
+
+    assert.expect(3);
+
+    argon2.hash("password", "somesalt", {
+      timeCost: -4294967290
+    }, function (err, hash) {
+      assert.ok(err, "Error should be defined.");
+      assert.equal(err.message, "Time cost must be positive.");
+      assert.equal(undefined, hash);
+      assert.done();
+    });
+  },
+
+  test_hash_low_time_cost: function (assert) {
+    "use strict";
+
+    assert.expect(3);
+
+    argon2.hash("password", "somesalt", {
+      timeCost: 0
+    }, function (err, hash) {
+      assert.ok(err, "Error should be defined.");
+      assert.equal(err.message, "Time cost is too small");
+      assert.equal(undefined, hash);
+      assert.done();
+    });
+  },
+
+  test_hash_high_time_cost: function (assert) {
+    "use strict";
+
+    assert.expect(3);
+
+    argon2.hash("password", "somesalt", {
+      timeCost: 4294967297
+    }, function (err, hash) {
```

Comment:
I chose 4294967297 because it becomes 1 when converted to uint32, while 4294967296 becomes 0. The value 1 is a valid input parameter for time cost, while 0 is not. Thus, if you remove the validation logic, the test fails with a more obviously concerning error when using 4294967297.

I was definitely inconsistent about it, though. Perhaps I should instead switch the other tests to ensure they overflow to valid inputs? Not a big deal either way. I could just switch them all to 4294967296 if you'd still prefer that.